scripts/src/main/resources/scripts/command/cobigen

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+# shellcheck source=scripts/functions
+source "$(dirname "${0}")"/../functions
+
+# $@: optional maven arguments
+function doSetup() {
+  # We need a JDK
+  doDevonCommand java -q setup
+  # We need Maven for installing plug-ins
+  doDevonCommand mvn -q setup
+  if [ -n "${1}" ] || [ ! -d "${COBIGEN_CLI_HOME}" ]
+  then
+    local software_version="LATEST"
+    local download_url="https://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.devonfw.cobigen&a=templates-devon4j&v=${software_version}"
+
+    doDownload "${download_url}" "${software_name}.jar" "${COBIGEN_CLI_HOME}"
+
+    # We need to extract correct batch files
+    if doIsWindows
+    then
+      doRunCommand "unzip -q '${COBIGEN_CLI_HOME}/${software_name}.jar' 'cobigen.bat' -d '${COBIGEN_CLI_HOME}'"
+      doRunCommand "unzip -q '${COBIGEN_CLI_HOME}/${software_name}.jar' 'cg.bat' -d '${COBIGEN_CLI_HOME}'"
+      doRunCommand "unzip -q '${COBIGEN_CLI_HOME}/${software_name}.jar' 'class-loader-agent.jar' -d '${COBIGEN_CLI_HOME}'"
+    else
+      doRunCommand "unzip -q '${COBIGEN_CLI_HOME}/${software_name}.jar' 'cobigen.sh' -d '${COBIGEN_CLI_HOME}'"
+      doRunCommand "unzip -q '${COBIGEN_CLI_HOME}/${software_name}.jar' 'cg.sh' -d '${COBIGEN_CLI_HOME}'"
+      doRunCommand "unzip -q '${COBIGEN_CLI_HOME}/${software_name}.jar' 'class-loader-agent.jar' -d '${COBIGEN_CLI_HOME}'"
+    fi
+
```

Comment:
The "doIsWindows" is a bit misleading name: It (currently) results in "yes", when the local bash environment is either msys (typically installed by git-for-windows) or cygwin. But both those firewalls need the *.sh files, when cobigen shall be started from **inside** the environment. In addition to that, the *.bat may also help - but only, when a developer also uses windows-CMD.

It might be easiest to extract just all files in any case.

Reply:
Yes, when we install maven or other multi-platform tools, we just extract everything.
Cant you just use `doInstall` function for cobigen as well?

Reply:
Unfortunately I cannot use `doInstall` as my `.jar` extension is not on the list of `doExtract`.

Reply:
I can support .jar in `doExtract` in a second. Why do you have "useless" files in your CLI release?
Can you change the way you bundle the release? (we do not have to block this PR now, but on the long run I would prefer cleaning this code as it appears odd to me).
Further the content of this JAR is designed/defined in CobiGen but here you are spreading redundant knowledge about it. What if some day CobiGen decides to rename or replace `class-loader-agent.jar`? Then this will break and nobody will notice as the CobiGen team is not anymore aware about this dependency in another repo...